packages/backend-function/src/function_env_type_generator.test.ts

```diff
@@ -0,0 +1,46 @@
+import { beforeEach, describe, it, mock } from 'node:test';
+import fs from 'fs';
+import { FunctionEnvironmentTypeGenerator } from './function_env_type_generator.js';
+import assert from 'assert';
+
+void describe('functionEnvironmentTypeGenerator', () => {
+  const fdCloseMock = mock.fn();
+  const fsOpenSyncMock = mock.method(fs, 'openSync');
+  const fsMkdirSyncMock = mock.method(fs, 'mkdirSync', () => null);
+  const fsWriteFileSyncMock = mock.method(fs, 'writeFileSync', () => null);
+
+  beforeEach(() => {
+    fdCloseMock.mock.resetCalls();
+    fsOpenSyncMock.mock.resetCalls();
+    fsMkdirSyncMock.mock.resetCalls();
+    fsWriteFileSyncMock.mock.resetCalls();
+  });
+
+  void it('generates a type definition file', () => {
```

Comment:
we should also have a test that actually writes the file and imports it to verify that it is valid syntax. I think it's okay for that to be a unit test even though it's doing some file i/o because it should be very fast

Reply:
Ok I wasn't sure about this one which was why I just tested if one of the static env vars is passed to `writeFileSync`